### PR TITLE
refactor: refine like optimistic update

### PR DESCRIPTION
**Pull request by swkatmask**
MF-0000 FW-0000

### CODE REVIEW

File: src/app/(normal)/following/page.tsx [hunk 1]
```diff
@@ -31,10 +31,11 @@ export default function Following() {
 
     const { data, hasNextPage, fetchNextPage, isFetchingNextPage, isFetching } = useSuspenseInfiniteQuery({
         queryKey: [
-            'following',
+            'posts',
```
Comment (guanbinrui) [inline src/app/(normal)/following/page.tsx:34]: Those keys should be constants.

File: src/components/Actions/Like.tsx [hunk 2]
```diff
@@ -70,67 +80,59 @@ export const Like = memo<LikeProps>(function Like({ count, hasLiked, postId, aut
...
+            enqueueSuccessMessage(hasLiked ? t`Unliked` : t`Liked`);
             togglePostLikeQueryData(queryClient, source, postId);
 
             return;
         } catch (error) {
             if (error instanceof Error) {
-                setRealCount(originalCount);
                 enqueueErrorMessage(
-                    liked ? t`Failed to unlike. ${error.message}` : t`Failed to like. ${error.message}`,
+                    hasLiked ? t`Failed to unlike. ${error.message}` : t`Failed to like. ${error.message}`,
```
Comment (guanbinrui) [inline src/components/Actions/Like.tsx:96]:
```suggestion
                    hasLiked ? t`Failed to unlike.` : t`Failed to like.`,
```

File: src/components/Actions/Like.tsx [hunk 3]
```diff
@@ -58,8 +64,12 @@ function togglePostLikeQueryData(queryClient: QueryClient, source: SocialPlatfor
 export const Like = memo<LikeProps>(function Like({ count, hasLiked, postId, authorId, source, disabled = false }) {
     const isLogin = useIsLogin(source);
     const queryClient = useQueryClient();
-    const [liked, setLiked] = useState(hasLiked);
-    const [realCount, setRealCount] = useState(count);
+
+    const currentProfileAll = useCurrentProfileAll();
+    const orderedProfileIds = useMemo(
```
Comment (guanbinrui) [inline src/components/Actions/Like.tsx:69]: Didn't use.

File: src/components/Actions/Like.tsx [hunk 4]
```diff
@@ -30,24 +32,28 @@ interface LikeProps {
...
         if (!old) return old;
 
         return produce(old, (draft) => {
-            LOOP: for (const page of draft.pages) {
+            for (const page of draft.pages) {
                 for (const post of page.data) {
                     if (post.postId === postId) {
                         post.hasLiked = !post.hasLiked;
-                        break LOOP;
+                        post.stats = {
```
Comment (guanbinrui) [inline src/components/Actions/Like.tsx:50]: Should we do the same thing after doing other reactions?

File: src/app/(normal)/following/page.tsx (hunk 1 above)
Comment (swkatmask) [reply src/app/(normal)/following/page.tsx:34]: Do not get it. react-query doesn't have such a limitation.

File: src/components/Actions/Like.tsx (hunk 4 above)
Comment (swkatmask) [reply src/components/Actions/Like.tsx:50]: Maybe.

File: src/app/(normal)/following/page.tsx (hunk 1 above)
Comment (swkatmask) [reply src/app/(normal)/following/page.tsx:34]: The query filter `{ queryKey: ['posts', source] }` used in `setQueriesData` will match all queries of post list.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**